src/request_handlers/shop.rs

```diff
@@ -40,19 +43,24 @@ enum ShopRequest {
 #[serde(tag = "type")]
 pub enum ShopResponse {
     DownloadOkay {
-        id: i32,
+        /// app_id of the downloaded app.
+        app_id: String,
         /// Name to be used as filename in `sendToChat`.
         name: String,
         /// Base64 encoded webxdc.
         data: String,
     },
     DownloadError {
-        id: i32,
+        app_id: String,
         error: String,
     },
     Update {
+        /// List of new app ids.
         app_infos: Vec<AppInfo>,
+        /// List of removed [AppInfo] ids.
         serial: i32,
```

Comment:
AppInfo are objects about a specific version of an app but the actual deletion code seems to assume it's an `app_id` but then the comment here is misleading/wrong.  Also not sure how the "removed" list is coming from the server.

Reply:
I updated the comments about this one as they were in fact wrong.